src/import/protobuf-c/library/protobuf-c.c

```diff
@@ -1863,6 +1866,7 @@ repeated_field_pack_to_buffer(const ProtobufCFieldDescriptor *field,
 		rv += uint32_pack(payload_len, scratch + rv);
 		buffer->append(buffer, rv, scratch);
 		tmp = pack_buffer_packed_payload(field, count, array, buffer);
+		(void) tmp; /* Warning fix by Xively */
 		assert(tmp == payload_len);
```

Comment:
Is this only ESP32 specific problem? I guess yet.
Can't we solve this by defining assert something like this:
#define assert( x ) (void)x
? 
This way a third party code could stay intact and a version refresh wouldn't break the ESP32 break.

Reply:
What do you mean by 'an ESP32 specific file'? I've tried modifying xi_debug.h to `#define assert( e ) ( ( void )(e) )` instead of `#define assert( e ) ( ( void )0 )`, but that didn't fix the problem. I think the ESP's SDK may be overwriting the macro, but I'm not sure.

I'd also like to avoid modifying 3rd party code, but I don't think this particular change is worth the time. Let me know if you disagree.